src/Umbraco.Core/Persistence/PetaPoco.cs

```diff
@@ -224,13 +226,18 @@ private void CommonConstruct()
 		// Automatically close one open shared connection
 		public void Dispose()
 		{
-			// Automatically close one open connection reference
-			//  (Works with KeepConnectionAlive and manually opening a shared connection)
-			CloseSharedConnection();
+		    Dispose(true);

```

Comment:
We have no unmanaged resources to release here so I don't think we need a finalizer. The underlying database connection is a managed object (which in turns might have unmanaged resources).